openedx/core/djangoapps/password_policy/utils.py

```diff
@@ -0,0 +1,80 @@
+import pytz
+from datetime import datetime, timedelta
+from dateutil.parser import parse as parse_date
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.models import Group
+from django.utils.translation import ugettext as _
+
+from openedx.core.djangoapps.util.user_messages import PageLevelMessages, UserMessageType
+from openedx.core.djangoapps.password_policy.constants import PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME
+from util.date_utils import strftime_localized, DEFAULT_SHORT_DATE_FORMAT
+from util.password_policy_validators import ValidationError, validate_password
+
+
+class NonCompliantPasswordException(Exception):
+    pass
+
+
+def check_password_policy_compliance(user, password, request=None):
+    config = settings.PASSWORD_POLICY_COMPLIANCE_ROLLOUT_CONFIG
+    if not (config and config.get('ENABLE_COMPLIANCE_CHECKING', False)):
+        return
+
+    # If the user is already known to be compliant, return.
+    if user.groups.filter(name=PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME).exists():
+        return
+
+    is_compliant = is_password_compliant(user, password)
+    if is_compliant:
+        user.groups.add(Group.objects.get(name=PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME))
+        user.save()
+        return
+
+    deadline = get_enforcement_deadline_for_user(config, user)
+    if deadline is None:
+        return
+
+    now = datetime.now(pytz.UTC)
+    warning_days = config.get('DAYS_BEFORE_DEADLINE_TO_SHOW_WARNING', 0)
+
+    if now >= deadline:
+        raise NonCompliantPasswordException()
+    elif request and warning_days and deadline - timedelta(days=warning_days) <= now:
+        messages.add_message(
+            request,
+            UserMessageType.WARNING.value,
+            _('Your password is not compliant with current policy requirements and will need to be updated on or '
+              'before {deadline}').format(deadline=strftime_localized(deadline, DEFAULT_SHORT_DATE_FORMAT)),
+            extra_tags=PageLevelMessages.get_namespace()
+        )
```

Comment:
This should be cleaned up before merge, as it re-implements behavior from the PageLevelMessages class so that we can avoid including HTML in the message.